### PR TITLE
[#4591] Add Integrity check for books with edition reported as 1

### LINKED ISSUE #4591: Add check for books with first edition
**Is your suggestion for improvement related to a problem? Please describe.**
Some journals have the convention to only print the edition of a book if it's not the first. (It's hard to find a reliable source on this, but see e.g [here on p. 19](https://www.otago.ac.nz/library/pdf/Harvard_referencing.pdf))

**Describe the solution you'd like**
Have an integrity check that reports books with `edition = 1`

### PR DESCRIPTION
**Pull request by RonakLakhotia**
Fixes #4591

### CODE REVIEW

**Comment by RonakLakhotia** [comment]
@tobiasdiez @Siedlerchr could you review this pr, also I am not sure why travis is failing

**Comment by Siedlerchr** [comment]
@RonakLakhotia  If you click on the travis build failed (Details)  you can expand the failing job, in your case it's just the localization test which failed.  You need to add the string to the english localization file under src/main/resources/l10n

```
org.jabref.logic.l10n.LocalizationConsistencyTest > findMissingLocalizationKeys() FAILED
    org.opentest4j.AssertionFailedError: DETECTED LANGUAGE KEYS WHICH ARE NOT IN THE ENGLISH LANGUAGE FILE
    PASTE THESE INTO THE ENGLISH LANGUAGE FILE
    edition\ of\ book\ reported\ as\ just\ 1=edition of book reported as just 1
     ==> expected: <[]> but was: <[edition\ of\ book\ reported\ as\ just\ 1 (src/main/java/org/jabref/logic/integrity/EditionChecker.java LANG)]>
        at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
        at org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:195)
        at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:184)
        at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1135)
        at org.jabref.logic.l10n.LocalizationConsistencyTest.findMissingLocalizationKeys(LocalizationConsistencyTest.java:111)
```

**Comment by RonakLakhotia** [comment]
Thanks for the pointer! Fixed the failing test. Ready for review

**Comment by RonakLakhotia** [comment]
@tobiasdiez ready for review

File: src/main/java/org/jabref/logic/integrity/EditionChecker.java [hunk 1]
```diff
@@ -44,10 +45,15 @@ public EditionChecker(BibDatabaseContext bibDatabaseContext) {
         }
 
         //BibTeX
-        if (!bibDatabaseContextEdition.isBiblatexMode() && !FIRST_LETTER_CAPITALIZED.test(value.trim())) {
+        if (!bibDatabaseContextEdition.isBiblatexMode() && !FIRST_LETTER_CAPITALIZED.test(value.trim()) &&
+                !value.equals(FIRST_EDITION)) {
             return Optional.of(Localization.lang("should have the first letter capitalized"));
         }
 
+        if (value.equals(FIRST_EDITION)) {
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/logic/integrity/EditionChecker.java]: Personally, I would move this check before the biblatex/bibtex checks (then you don't need to modify the conditions of these if statements).

Comment (RonakLakhotia) [reply src/main/java/org/jabref/logic/integrity/EditionChecker.java]: @tobiasdiez yes, it makes more sense. Updated PR.

**Comment by RonakLakhotia** [comment]
I am not completely sure of the build failure errors? Any pointers?

**Comment by RonakLakhotia** [comment]
Update, there are unresolved dependencies . `./gradlew -q checkOutdatedDependencies` is there a command to update them? @tobiasdiez

**Comment by Siedlerchr** [comment]
You can ignore the outdated dependencies. It's one of the tasks which is allowed to fail.

**Comment by RonakLakhotia** [comment]
Build passed, Ready for review !

**Comment by RonakLakhotia** [comment]
Any updates?

**Comment by tobiasdiez** [comment]
@RonakLakhotia We have the policy that at least two core developers have to approve a pull request before me merge. Since we are all working on JabRef in our freetime, reviewing of a PR might take a day or two (especially over the weekend). 

But that shouldn't stop you from looking and working at other open issues if you have the motivation and the drive to do so!

**Comment by RonakLakhotia** [comment]
@tobiasdiez sure, I am new to open source hence I thought I should get one PR merged before going ahead

**Pull request merged**